### PR TITLE
Add DAC1 & DAC2 in stm32f3

### LINKED ISSUE #35992: stm32f303k8 device tree missing DACs
Hi,

I'm developing with a stm32f303k8 and I've found a few items missing from the device tree. These are both DAC peripherals (DAC1 and DAC2), CAN alternate pins (PB8 and PB9). I also had some issues with timer3 pwm channels.

I'm making a custom board which includes
```
#include <st/f3/stm32f303X8.dtsi>
#include <st/f3/stm32f303k(6-8)tx-pinctrl.dtsi>
```
which further intern includes 
```
#include <st/f3/stm32f3.dtsi>
#include <st/f3/stm32f303.dtsi>
```

Firstly both DACs are missing:
```
/ {
       soc {
		/*
		 * The DACs are not down as being in the included
		 * device trees, so we need full details of them
		 */
		dac1: dac@40007400 {
			compatible = "st,stm32-dac";
			reg = <0x40007400 0x400>; // At location, with length
			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x20000000>; // clock, and bit that enables it
			status = "disabled";
			label = "DAC_1";
			#io-channel-cells = <1>;
		};
		
		dac2: dac@40009800 {
			compatible = "st,stm32-dac";
			reg = <0x40009800 0x400>;
			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x04000000>;
			status = "disabled";
			label = "DAC_2";
			#io-channel-cells = <1>;
		};
        };
};
```

Secondly there are pin-muxes missing from #include <st/f3/stm32f303k(6-8)tx-pinctrl.dtsi>
```
/ {
       soc {
		pinctrl: pin-controller@48000000 {
			// The below seem to not be in the include (yet are in the datasheet)
			can1_rx_pb8: can1_rx_pb8 {
				pinmux = <STM32_PINMUX('B', 8, AF9)>;
				bias-pull-up;
			};
			can1_tx_pb9: can1_tx_pb9 {
				pinmux = <STM32_PINMUX('B', 9, AF9)>;
			};
		};
        };
};
```

I also couldn't get my the timer3 pwm pinmuxes to come through so I had to add those to the pinctrl:
```
/ {
       soc {
		pinctrl: pin-controller@48000000 {
			// These seemed to not be included
			tim3_ch2_pwm_pa7: tim3_ch2_pwm_pa7 {
				pinmux = <STM32_PINMUX('A', 7, AF2)>;
			};
			tim3_ch3_pwm_pb0: tim3_ch3_pwm_pb0 {
				pinmux = <STM32_PINMUX('B', 0, AF2)>;
			};
			tim3_ch4_pwm_pb1: tim3_ch4_pwm_pb1 {
				pinmux = <STM32_PINMUX('B', 1, AF2)>;
			};
		};
        };
};
```

### PR DESCRIPTION
**Pull request by ikmdani**
These commits adds DAC1 support in dtsi for stm32f3 series. Some
stm32f3 series has DAC2 nodes for in dtsi for the same where applicable.
DAC2 is present in stm32f303x8 and stm32f373xc.

Fixes #35992

### CODE REVIEW

**Review by FRASTM — APPROVED** [review]

**Review by ABOSTM — CHANGES_REQUESTED** [review]
DAC2 is also present in stm32f334x8.
DAC support should be added to doc/index.rst for stm32f3_disco

**Comment by ikmdani** [comment]
> DAC2 is also present in stm32f334x8.

DAC2 is present in stm32f378xx, stm32f303x8, stm32f373xc, stm32f328xx, stm32f334x8. I have added DAC2 nodes in stm32f303x8.dtsi and stm32f373xc.dtsi only as they exist. For other 3 platforms stm32f378xx.dtsi, stm32f328xx.dtsi and stm32f334x8.dtsi does not exist so i did not add. Pls let me know if there is any other way i can add.

File: boards/arm/stm32f3_disco/stm32f3_disco.dts [hunk 1]
```diff
@@ -187,3 +187,8 @@
 	pinctrl-0 = <&adc1_in1_pa0>;
 	status = "okay";
 };
+
+&dac1 {
+	status = "okay";
+	pinctrl-0 = <&dac_out1_pa4>;
```
Comment (FRASTM) [inline boards/arm/stm32f3_disco/stm32f3_disco.dts:193]: PA4 will probably conflict with 
&spi1 {
	pinctrl-0 = <&spi1_nss_pa4 &spi1_sck_pa5
		     &spi1_miso_pa6 &spi1_mosi_pa7>;
	status = "okay";
};

**Review by FRASTM — APPROVED** [review]

**Comment by ikmdani** [comment]
> I agree that DAC should be added only when device dtsi exist.
> But file zephyr\dts\arm\st\f3\stm32f334X8.dtsi exist

You are right, i missed to check the last dtsi, thanks for pointing this out.

File: boards/arm/stm32f3_disco/doc/index.rst [hunk 2]
```diff
@@ -105,6 +105,8 @@ features:
 +-----------+------------+-------------------------------------+
 | ADC       | on-chip    | adc                                 |
 +-----------+------------+-------------------------------------+
+| DAC       | on-chip    | dac                                 |
```
Comment (erwango) [inline boards/arm/stm32f3_disco/doc/index.rst:108]: This change should be moved to dedicated board commit

File: boards/arm/stm32f3_disco/stm32f3_disco.dts (hunk 1 above)
Comment (erwango) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts:193]: @ikmdani Is there another pin that could be available to avoid this conflct?

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ikmdani** [comment]
> @ikmdani Is there another pin that could be available to avoid this conflct?

Referring to the RM0316 document, section 16.2, below is the note on DACx Pins.
```
Note: Once the DACx channel y is enabled, the corresponding GPIO pin (PA4, PA5 or PA6) is
automatically connected to the analog converter output (DACx_OUTy). In order to avoid
parasitic consumption, the PA4, PA5 or PA6 pin should first be configured to analog (AIN).
```

And PA5 & PA6 are already used as
```
&spi1 {
        pinctrl-0 = <&spi1_nss_pa4 &spi1_sck_pa5
                     &spi1_miso_pa6 &spi1_mosi_pa7>;
        status = "okay";
};
```

Kindly suggest.

**Comment by erwango** [comment]
> the RM0316 document, section 16.2, below is the note on DACx Pins.
> 
> ```
> Note: Once the DACx channel y is enabled, the corresponding GPIO pin (PA4, PA5 or PA6) is
> automatically connected to the analog converter output (DACx_OUTy). In order to avoid
> parasitic consumption, the PA4, PA5 or PA6 pin should first be configured to analog (AIN).
> ```

Ok, let's keep it that way then, but please add a comment highlighting the conflict in the file directly

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**